typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
```

Comment:
Discussion for recommended rule: `'@typescript-eslint/no-unnecessary-type-assertion': 'error'`

Reply:
The fixer for this rule didn't work for me. It left extra `<` character when removing a `<Conversion>`. I filed a bug but I'm curious if anyone else sees this. https://github.com/typescript-eslint/typescript-eslint/issues/3196

Reply:
The bug was fixed in 4.21.0 which released a couple days ago so we should upgrade from our current 4.17.0 at some point. https://github.com/typescript-eslint/typescript-eslint/blob/master/CHANGELOG.md

Reply:
As it sounds like the issue is only in the fixer and not in the rule we can do the version update in a follow-up PR.
Proposal: Do not disable the recommended rule.